File: app/src/main/java/com/cottondroid/todoachieved/task/form/time/TimeTextView.java
package com.cottondroid.todoachieved.task.form.time;

import android.content.Context;
import android.util.AttributeSet;

import com.cottondroid.todoachieved.R;

import java.text.DateFormat;
import java.util.Calendar;

public class TimeTextView extends DateTimeTextView {

    public TimeTextView(Context context) {
        super(context);
    }

    public TimeTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public TimeTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public DateFormat getDateFormat() {
        return DateFormat.getTimeInstance();
    }

    public void showDateTime(Calendar calendar) {
        if (calendar.get(Calendar.HOUR_OF_DAY) == 0 && calendar.get(Calendar.MINUTE) == 0) {
            setText(getDefaultText());
        } else {
            super.showDateTime(calendar);
        }
    }

    @Override
    public int getDefaultText() {
        return R.string.task_form_choose_time;
    }

}

File: app/src/main/java/com/cottondroid/todoachieved/task/form/time/TimeTextView.kt
package com.cottondroid.todoachieved.task.form.time

import android.content.Context
import android.util.AttributeSet
import com.cottondroid.todoachieved.R
import java.text.DateFormat
import java.util.Calendar

class TimeTextView(context: Context, attrs: AttributeSet) : DateTimeTextView(context, attrs) {
    override val dateFormat: DateFormat
        get() = DateFormat.getTimeInstance()

    override fun showDateTime(calendar: Calendar) {
        if (calendar[Calendar.HOUR_OF_DAY] == 0 && calendar[Calendar.MINUTE] == 0) {
            setText(defaultText)
        } else {
            super.showDateTime(calendar)
        }
    }

    override val defaultText: Int
        get() = R.string.task_form_choose_time
}